homeassistant/components/mqtt/client.py

```diff
@@ -897,6 +904,9 @@ async def _wait_for_mid(self, mid: int) -> None:
                 self._pending_operations_condition.notify_all()
 
     async def _discovery_cooldown(self) -> None:
+        """Wait until all discovery and subscriptions are processed."""
+        # Await DISCOVERY_COOLDOWN first
+        await asyncio.sleep(INITIAL_SUBSCRIBE_COOLDOWN)
```

Comment:
Why is this sleep added?

Reply:
Because it will take at least `INITIAL_SUBSCRIBE_COOLDOWN` from the first subscribe to the first actual subscribe during setup.

Reply:
True